packages/docusaurus-mdx-loader/src/remark/headings/index.js

```diff
@@ -5,12 +5,14 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-/* Based on remark-slug (https://github.com/remarkjs/remark-slug) */
+/* Based on remark-slug (https://github.com/remarkjs/remark-slug) and gatsby-remark-autolink-headers (https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-autolink-headers) */
 
 const visit = require('unist-util-visit');
 const toString = require('mdast-util-to-string');
 const slugs = require('github-slugger')();
 
+const customHeadingIdRegex = /^(.*?)\s*\{#([\w-]+)\}$/;
```

Comment:
What about extracting `customHeadingIdRegex.exec(heading);` in a separate function `const {heading,id} = extractHeadingId(rawHeading)` and adding tests? 

That would simplify the testing of Regepx edge cases.

Reply:
Hmm, but we have separate test that with different cases, so it seems to me redundant extract new function (and its test), because in many ways it will duplicate this test.